qa/L0_video_input_decoupled/client.py

```diff
@@ -121,14 +119,28 @@ def parse_args():
                     if this_id not in result_dict.keys():
                         result_dict[this_id] = []
                     result_dict[this_id].append(data_item)
+
+                expected_result, = ref_pipe.run()
+                if args.device == 'gpu':
+                    expected_result = expected_result.as_cpu().as_array()
+                else:
+                    expected_result = expected_result.as_array()
+                if this_id not in expected_dict.keys():
+                    expected_dict[this_id] = []
+                expected_dict[this_id].append(expected_result)
+
                 recv_count += 1
 
             result_list = result_dict[request_id]
-            expected_result = np.split(expected_result, n_frames / FRAMES_PER_SEQUENCE)
-            for i, result in enumerate(result_list):
-                expected_batch = expected_result[i * BATCH_SIZE : min((i+1) * BATCH_SIZE, len(expected_result))]
-                expected_batch = np.asarray(expected_batch)
-                result_data = result.as_numpy('OUTPUT')
-                assert np.allclose(expected_batch, result_data)
+            expected_list = expected_dict[request_id]
+            for res, exp in zip(result_list, expected_list):
+                res = res.as_numpy("OUTPUT")
+                if not np.allclose(res, exp):
```

Comment:
```suggestion
                if not np.allclose(res, exp, atol=1):
```
?

Reply:
I'd like to know if it fails. `atol` wouldn't be that useful, due to nature of the error. Sometimes it's 255...